test/common.ts

```diff
@@ -69,87 +76,96 @@ export const facility: Facility = {
-  maxNumberOfAdultOccupantsOneof: {
-    oneofKind: 'maxNumberOfAdultOccupants',
-    maxNumberOfAdultOccupants: 2
-  },
-  maxNumberOfChildOccupantsOneof: {
-    oneofKind: 'maxNumberOfChildOccupants',
-    maxNumberOfChildOccupants: 2
-  },
-  maxNumberOfOccupantsOneof: {
-    oneofKind: 'maxNumberOfOccupants',
-    maxNumberOfOccupants: 2
-  },
-  privateHomeOneof: { oneofKind: 'privateHome', privateHome: false },
-  suiteOneof: { oneofKind: 'suite', suite: false },
-  tierOneof: { oneofKind: 'tier', tier: SpaceTier.DEFAULT_STANDARD },
-  views: {
-    viewOfValleyOneof: { oneofKind: 'viewOfValley', viewOfValley: true },
-    viewOfBeachOneof: { oneofKind: 'viewOfBeach', viewOfBeach: false },
-    viewOfCityOneof: { oneofKind: 'viewOfCity', viewOfCity: false },
-    viewOfGardenOneof: { oneofKind: 'viewOfGarden', viewOfGarden: false },
-    viewOfLakeOneof: { oneofKind: 'viewOfLake', viewOfLake: false },
-    viewOfLandmarkOneof: { oneofKind: 'viewOfLandmark', viewOfLandmark: false },
-    viewOfOceanOneof: { oneofKind: 'viewOfOcean', viewOfOcean: false },
-    viewOfPoolOneof: { oneofKind: 'viewOfPool', viewOfPool: false }
-  },
-  totalLivingAreas: {
-    sleeping: {
-      numberOfBedsOneof: { oneofKind: 'numberOfBeds', numberOfBeds: 1 },
-      kingBedsOneof: { oneofKind: 'kingBeds', kingBeds: 1 },
-      queenBedsOneof: {
-        oneofKind: 'queenBedsException',
-        queenBedsException: Exception.UNSPECIFIED_REASON
-      },
-      doubleBedsOneof: {
-        oneofKind: 'doubleBedsException',
-        doubleBedsException: Exception.UNSPECIFIED_REASON
-      },
-      singleOrTwinBedsOneof: {
-        oneofKind: 'singleOrTwinBedsException',
-        singleOrTwinBedsException: Exception.UNSPECIFIED_REASON
-      },
-      bunkBedsOneof: {
-        oneofKind: 'bunkBedsException',
-        bunkBedsException: Exception.UNSPECIFIED_REASON
-      },
-      sofaBedsOneof: {
-        oneofKind: 'sofaBedsException',
-        sofaBedsException: Exception.UNSPECIFIED_REASON
-      },
-      otherBedsOneof: { oneofKind: 'otherBeds', otherBeds: 0 },
-      cribsOneof: { oneofKind: 'cribs', cribs: false },
-      cribsAvailableOneof: {
-        oneofKind: 'cribsAvailableException',
-        cribsAvailableException: Exception.UNSPECIFIED_REASON
-      },
-      cribCountOneof: {
-        oneofKind: 'cribCountException',
-        cribCountException: Exception.UNSPECIFIED_REASON
-      },
-      rollAwayBedsOneof: { oneofKind: 'rollAwayBeds', rollAwayBeds: false },
-      rollAwayBedsAvailableOneof: {
-        oneofKind: 'rollAwayBedsAvailableException',
-        rollAwayBedsAvailableException: Exception.UNSPECIFIED_REASON
-      },
-      rollAwayBedCountOneof: {
-        oneofKind: 'rollAwayBedCountException',
-        rollAwayBedCountException: Exception.UNSPECIFIED_REASON
+export const space: Item = {
+  name: 'Mountain view room',
+  description: 'Panoramic views of Ushba',
+  photos: [],
+  type: ItemType.SPACE,
+  payload: Space.toBinary({
```

Comment:
I propose to leave Space in json format, validate it and then encode it on the server side